_documentation/04-packaging-drivers.md

```diff
@@ -1,4 +1,295 @@
+
+___
+
+* **base**
+<br>The base keyword declares which base snap to use with your project. A base snap is a special kind of snap that provides a run-time environment alongside a minimal set of libraries that are common to most applications:
+
+        base: core18
+
+    As used above, core18 is the current standard base for snap building and is based on Ubuntu 18.04 LTS.
+
+    See <a href="https://snapcraft.io/docs/base-snaps">Base snaps</a> for more details.
+
+___
+
+* **version**
+<br>A human-readable string(enclosed within single-quotes) of maximum size 32 characters, represents the version number of your snap. A higher version number generally corresponds to a new application. 
+
+        version: '1.0'
+
+___
+
+* **summary**
+<br>A single line summary that maybe 79 characters long, describes the snap in short and simple terms. It is used when users are searching through the Store for your application.
+
+        summary: HP Printer Application
+
+___
+
+* **description**
+<br>Description is used to provide a little more detail about your application. You could as many lines as you want in this field.
+
+        description: |
+            HP Printer Application is a PAPPL (Printer Application Framework) based printer application
+            to support hp printers.
+            PAPPL is a simple C-based framework/library for developing CUPS
+            Printer Applications, which are the recommended replacement for
+            printer drivers.
+
+___
+
+* **grade**
+<br> Grade defines the quality grade of the snap. It can have two values.
+    
+    * **devel**: a development version of the snap, so as not to be published to the stable or candidate channels.
+
+    * **stable**: a stable release or release candidate, which can be released to all channels.
+
+    We would be using the grade value as stable.
+
+        grade: stable
+
+___
+
+* **confinement**
+<br>Confinement determines if the snap should be restricted in access or not. It could have three possible values.
+
+    * **strict**: for no access outside of declared interfaces through plugs. Can be published easily and installed without any special command-line argument.
+    
+    * **devmode**: for unrestricted access to system resources. Snaps having this confinement cannot be published to Snap Store.
+    
+    * **classic**: Allows access to your system’s resources in much the same way traditional packages do. To safeguard against abuse, publishing a classic snap requires manual approval, and installation requires the --classic command-line argument.
+
+    We would be using the confinement value as strict.
+
+        confinement: strict
+
+___
+
+There are other metadata that could be used to provide more information to snapcraft besides the above-listed ones. One may refer to <a href="https://snapcraft.io/docs/snapcraft-top-level-metadata">Snapcraft top-level format documentation</a> to know about these.
```

Comment:
There is other metadata ... "data" is never Plural.

Reply:
Thanks, Till, corrected.